server/src/main/java/org/elasticsearch/script/field/ShortDocValuesField.java

```diff
@@ -0,0 +1,140 @@
+    protected final SortedNumericDocValues input;
+    protected final String name;
+
+    protected short[] values = new short[0];
+    protected int count;
+
+    private ScriptDocValues.Longs longs = null;
+
+    public ShortDocValuesField(SortedNumericDocValues input, String name) {
+        this.input = input;
+        this.name = name;
+    }
+
+    /**
+     * Set the current document ID.
+     *
+     * @param docId
+     */
+    @Override
+    public void setNextDocId(int docId) throws IOException {
+        if (input.advanceExact(docId)) {
+            resize(input.docValueCount());
+            for (int i = 0; i < count; i++) {
+                values[i] = (short) input.nextValue();
+            }
+        } else {
+            resize(0);
+        }
+    }
+
+    protected void resize(int newSize) {
+        count = newSize;
+
+        assert count >= 0 : "size must be positive (got " + count + "): likely integer overflow?";
+        values = ArrayUtil.grow(values, count);
+    }
+
+    /**
+     * Returns a {@code ScriptDocValues} of the appropriate type for this field.
+     * This is used to support backwards compatibility for accessing field values
+     * through the {@code doc} variable.
+     */
+    @Override
+    public ScriptDocValues<Long> getScriptDocValues() {
+        if (longs == null) {
+            longs = new ScriptDocValues.Longs(this);
+        }
+
+        return longs;
+    }
+
+    @Override
+    public Long getInternal(int index) {
+        return (long) values[index];
+    }
+
+    /**
+     * Returns the name of this field.
+     */
+    @Override
+    public String getName() {
+        return name;
+    }
+
+    /**
+     * Returns {@code true} if this field has no values, otherwise {@code false}.
+     */
+    @Override
+    public boolean isEmpty() {
+        return count == 0;
+    }
+
+    /**
+     * Returns the number of values this field has.
+     */
+    @Override
+    public int size() {
+        return count;
+    }
+
+    /**
+     * Returns an iterator over elements of type {@code T}.
+     *
+     * @return an Iterator.
+     */
+    @Override
+    public Iterator<Short> iterator() {
+        return new Iterator<Short>() {
+            private int index = 0;
+
+            @Override
+            public boolean hasNext() {
+                return index < count;
+            }
+
+            @Override
+            public Short next() {
+                if (hasNext() == false) {
+                    throw new NoSuchElementException();
+                }
+                return values[index++];
+            }
+        };
+    }
+
+    public short get(int defaultValue) {
+        return get(0, defaultValue);
+    }
+
+    // constants in java and painless are ints, so letting the defaultValue be an int allows users to
+    // call this without casting. A short variable will be automatically widened to an int.
+    // If the user does pass a value outside the range, it will be cast down to a short.
```

Comment:
Since this is public, please convert this to a JavaDoc type comment.

Reply:
Changed to javadoc